src/store/currentTodo.ts

```diff
@@ -0,0 +1,36 @@
+import { Todo } from '../types/Todo';
+
+type SetTodoAction = {
+  type: 'SET_TODO',
+  payload: Todo,
+};
+
+type UnsetTodoAction = {
+  type: 'UNSET_TODO'
+};
+
+type Action = SetTodoAction | UnsetTodoAction;
+
+export const actions = {
+  setTodo: (todo: Todo): SetTodoAction => ({
+    type: 'SET_TODO',
+    payload: todo,
+  }),
+  unsetTodo: (): UnsetTodoAction => ({
+    type: 'UNSET_TODO',
+  }),
+};
+
+export const selectionTodoReducer = (
+  state: null | Todo = null,
+  action: Action,
+): Todo | null => {
+  switch (action.type) {
+    case 'SET_TODO':
+      return action.payload;
+    case 'UNSET_TODO':
+      return null;
+    default:
+      return state;
```

Comment:
You should return a copy of the state with changed keys.
Fix here and in the rest reducers

```suggestion
    case 'SET_TODO':
      return {
      	...state,
      	todos: action.payload,
      };
    case 'UNSET_TODO':
      return {
      	...state,
      	todos: null,
      };
    default:
      return state;
```

Reply:
in my case, I have no state with key todos, my state is todos and in this case it is unnecessary in my opinion